File: Programmers/Lv.1/덧칠하기/161989.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        
        int[] arr = new int[n];
        for (int num : section) {
            arr[num - 1] = 1;
        }
        
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 1) {
                for (int j = i; j < Math.min(i + m, arr.length); j++) {
                    arr[j] = 0;
                }
                answer++;
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/덧칠하기/161989.js
function solution(n, m, section) {
    let answer = 0;
    
    const arr = new Array(n).fill(0);
    section.forEach(v => arr[v - 1] = 1);
    
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === 1) {
            for (let j = i; j < Math.min(i + m, arr.length); j++) {
                arr[j] = 0;
            }
            answer++;
        }
    }
    
    return answer;
}
